### DISCUSSION TITLE
Supporting Zarr / NetCDF for XArray-type analyses

### DISCUSSION THREAD

**Discussion by MrPowers** [unitycatalog/unitycatalog#226] [Ideas]
Cool example of a XArray analysis:

<img width="545" alt="Screenshot 2024-07-13 at 1 41 34 PM" src="https://github.com/user-attachments/assets/4f5f8bbd-ced7-41a0-a813-886290707a95">

Link to [author's post](https://www.linkedin.com/posts/ryan-abernathey_xarray-activity-7216825104658259968-dfUQ?utm_source=share&utm_medium=member_desktop).

Is there a way to register these array datasets in Unity Catalog?

**Comment by rabernat** [comment]
Author here!

We have been studying Unity Catalog at [Earthmover](https://earthmover.io/) and are considering implementing it within our platform's data catalog.

Unity allows for either Tables or Volumes (for random unstructured data / files). But these datasets are neither: they are multidimensional array (tensor) datasets. Zarr datasets in particular are multi-file datasets which can span thousands of objects in object storage but represent a single coherent dataset, often modeled with [Xarray](https://xarray.dev/). These datasets are common in scientific computing but are not well supported in existing data lakehouse / warehouse soltuions.

Because they are logical datasets that span many individual files, they are closer to delta or iceberg tables than to Volumes. (It's not enough to just define a new file type in Volumes.) So ideally we would have a new type of entity to represent them at Schema level.

I understand that might be out of scope for Unity. But it's the most logical thing.
